assets/node-exporter/daemonset.yaml

```diff
@@ -10,6 +10,10 @@ spec:
   selector:
     matchLabels:
       app.kubernetes.io/name: node-exporter
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 10%
```

Comment:
Is unavailability of daemonsets during rollouts something that is expected? Why would daemonsets be unavailable systemically during a rolling update?

Reply:
The default for daemonsets is rollingUpdate with maxUnavailable 1, so there's already one unavailable in the current scheme, you can confirm by getting the daemonset from a cluster. This just changes it so that 10% of them may be unavailable at once which parallelizes the rollout, on a 250 node cluster that decreases rollout time from around 100 minutes down to under 10 minutes.

Your operator shouldn't change its behavior around marking itself upgraded / progressing. I guess if it's monitoring the number of unavailable that may need to change

Reply:
got it, so it is about parallelization

Reply:
10% seems like a good default